### PR TITLE
Module frontdoor

### PR DESCRIPTION
**Pull request by cliverallen**
# Change
Add a module for the front door service

## Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings
- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (readme)
- [X] I did format my code

[Network: Frontdoors](https://github.com/cliverallen/ResourceModules/actions/workflows/ms.network.frontdoors.yml)

### CODE REVIEW

File: arm/Microsoft.Network/frontDoors/.bicep/nested_rbac.bicep [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+param resourceId string
+
+var builtInRoleNames = {
+  'Owner': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')
+  'Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')
+  'Reader': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
+  'Avere Cluster Create': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'a7b1b19a-0e83-4fe5-935c-faaefbfd18c3')
+  'Avere Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '4f8fab4f-1852-4a58-a46a-8eaf358af14a')
+  'Azure Service Deploy Release Management Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '21d96096-b162-414a-8302-d8354f9d91b2')
+  'CAL-Custom-Role': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '7b266cd7-0bba-4ae2-8423-90ede5e1e898')
```
Comment (MrMCake) [inline arm/Microsoft.Network/frontDoors/.bicep/nested_rbac.bicep:13]: That seems to be a custom role. When you fetched the available roles - did you filter them by 'Built-In'?

File: arm/Microsoft.Network/frontDoors/readme.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+| `logsToEnable` | array | `[FrontdoorAccessLog, FrontdoorWebApplicationFirewallLog]` | `[FrontdoorAccessLog, FrontdoorWebApplicationFirewallLog]` | Optional. The name of logs that will be streamed. |
+| `metricsToEnable` | array | `[AllMetrics]` | `[AllMetrics]` | Optional. The name of metrics that will be streamed. |
+| `name` | string |  |  | Required. The name of the frontDoor. |
+| `roleAssignments` | array | `[]` |  | Optional. Array of role assignment objects that contain the 'roleDefinitionIdOrName' and 'principalId' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: '/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11' |
+| `routingRules` | array | `[]` |  | Required. Routing rules settings of the frontdoor resource. |
+| `sendRecvTimeoutSeconds` | int | `600` |  | Optional. Certificate name check time of the frontdoor resource. |
+| `tags` | object | `{object}` |  | Optional. Resource tags. |
+
+### Parameter Usage: `<ParameterPlaceholder>`
```
Comment (MrMCake) [inline arm/Microsoft.Network/frontDoors/readme.md:42]:
```suggestion
```

**Comment by MrMCake** [comment]
Hey @cliverallen, just 2 small comments. Thanks for the great contribution :)

File: arm/Microsoft.Network/frontDoors/.bicep/nested_rbac.bicep (hunk 1 above)
Comment (cliverallen) [reply arm/Microsoft.Network/frontDoors/.bicep/nested_rbac.bicep]: Tbh I copied this from the application gateway perms (https://github.com/cliverallen/ResourceModules/blob/main/arm/Microsoft.Network/applicationGateways/.bicep/nested_rbac.bicep) but have updated the file with just the builtin roles :)

File: arm/Microsoft.Network/frontDoors/readme.md (hunk 2 above)
Comment (cliverallen) [reply arm/Microsoft.Network/frontDoors/readme.md]: Line removed :)

**Review by MrMCake — APPROVED** [review]

**Pull request merged**